File: Multi/Maman14/server/models/request.cpp
//
// Created by HPC on 06/08/2024.
//

#include "request.h"

void to_json(nlohmann::json &j, const Request &r) {
    j = nlohmann::json{
            {"version",   r.version},
            {"type",      r.type},
            {"name_len",  r.name_len},
            {"file_name", r.file_name},
            {"size",      r.size},
            {"payload",   r.payload},

    };
}

void from_json(const nlohmann::json &j, Request &r) {

}

File: Multi/Maman14/client/models/request.py
from enum import Enum

from pydantic import BaseModel


class RequestTypes(Enum):
    DEFAULT = 111
    SAVE_TO_BACKUP = 100
    GET_FROM_BACKUP = 200
    REMOVE_FROM_BACKUP = 201
    GET_ALL_FILES_LIST = 202


class Request(BaseModel):
    user_id: str = ''
    version: str = ''
    type: RequestTypes = RequestTypes.DEFAULT
    name_len: int = 0
    file_name: str = ''
    size: int = 0
    payload: bytes = bytes()
